### PR TITLE
Bonding enhancements

### PR DESCRIPTION
**Pull request by narasimhan-v**

### CODE REVIEW

File: io/net/bonding.py [hunk 1]
```diff
@@ -292,12 +286,14 @@ def bond_setup(self, arg1, arg2):
...
                 cmd += 'ip link set %s down;' % val
             cmd += 'modprobe bonding;'
-            cmd += 'echo +%s > /sys/class/net/bonding_masters;'\
-                   % self.bond_name
-            cmd += 'echo 0 > /sys/class/net/%s/bonding/mode;'\
-                   % self.bond_name
-            cmd += 'echo 100 > /sys/class/net/%s/bonding/miimon;'\
-                   % self.bond_name
+            cmd += 'echo +%s > %s;'\
+                   % (self.bond_name, self.net_path)
```
Comment (harish-24) [inline io/net/bonding.py]: Is this not `(self.bond_name, self.bonding_masters_file)`?

**Comment by manvanthar** [comment]
Looks fine @narasimhan-v

**Comment by narasimhan-v** [comment]
Existing usage (still valid):
```
# avocado run bonding.py -m bonding.py.data/bonding_test.yaml --execution-order tests-per-variant
JOB ID     : c83294a5bf5900f731cae5e81e37d7bef46c8bb0
JOB LOG    : /root/Upstream/Everglade/results/job-2018-07-29T22.43-c83294a/job.log
 (1/6) bonding.py:Bonding.test_setup;active-backup-2640: PASS (12.28 s)
 (2/6) bonding.py:Bonding.test_run;active-backup-2640: PASS (87.26 s)
 (3/6) bonding.py:Bonding.test_cleanup;active-backup-2640: PASS (12.31 s)
 (4/6) bonding.py:Bonding.test_setup;802.3ad-1ac4: PASS (11.18 s)
 (5/6) bonding.py:Bonding.test_run;802.3ad-1ac4: PASS (87.22 s)
 (6/6) bonding.py:Bonding.test_cleanup;802.3ad-1ac4: PASS (15.34 s)
RESULTS    : PASS 6 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 225.88 s
JOB HTML   : /root/Upstream/Everglade/results/job-2018-07-29T22.43-c83294a/results.html
# 
```

With setup and cleanup:
```
# avocado run bonding.py:Bonding.test_setup -m bonding.py.data/bonding_single.yaml
JOB ID     : ae9bc43045922c587533c9d6307247761207c6c4
JOB LOG    : /root/Upstream/Everglade/results/job-2018-07-29T22.56-ae9bc43/job.log
 (1/1) bonding.py:Bonding.test_setup;run-bdf8: PASS (11.27 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 11.36 s
JOB HTML   : /root/Upstream/Everglade/results/job-2018-07-29T22.56-ae9bc43/results.html

# ip addr show bondtest
50: bondtest: <BROADCAST,MULTICAST,MASTER,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether c6:2c:d6:f5:3e:04 brd ff:ff:ff:ff:ff:ff
    inet 175.175.68.11/24 scope global bondtest
       valid_lft forever preferred_lft forever
    inet6 2002:903:15f:550:31a9:d133:2f8f:4be3/64 scope global temporary dynamic 
       valid_lft 604790sec preferred_lft 85790sec
    inet6 2002:903:15f:550:c42c:d6ff:fef5:3e04/64 scope global mngtmpaddr dynamic 
       valid_lft 2591990sec preferred_lft 604790sec
    inet6 fe80::c42c:d6ff:fef5:3e04/64 scope link 
       valid_lft forever preferred_lft forever

# avocado run bonding.py:Bonding.test_run -m bonding.py.data/bonding_single.yaml
JOB ID     : dbe0c62311824ab1cc4e5201d6888c81b9aca919
JOB LOG    : /root/Upstream/Everglade/results/job-2018-07-29T22.57-dbe0c62/job.log
 (1/1) bonding.py:Bonding.test_run;run-bdf8: PASS (87.26 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 87.36 s
JOB HTML   : /root/Upstream/Everglade/results/job-2018-07-29T22.57-dbe0c62/results.html

# ip addr show bondtest
50: bondtest: <BROADCAST,MULTICAST,MASTER,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether c6:2c:d6:f5:3e:04 brd ff:ff:ff:ff:ff:ff
    inet 175.175.68.11/24 scope global bondtest
       valid_lft forever preferred_lft forever
    inet6 2002:903:15f:550:31a9:d133:2f8f:4be3/64 scope global temporary dynamic 
       valid_lft 604686sec preferred_lft 85686sec
    inet6 2002:903:15f:550:c42c:d6ff:fef5:3e04/64 scope global mngtmpaddr dynamic 
       valid_lft 2591886sec preferred_lft 604686sec
    inet6 fe80::c42c:d6ff:fef5:3e04/64 scope link 
       valid_lft forever preferred_lft forever

# avocado run bonding.py:Bonding.test_cleanup -m bonding.py.data/bonding_single.yaml
JOB ID     : 0c254e316d16c1c2df4435f78e57be1e029beeeb
JOB LOG    : /root/Upstream/Everglade/results/job-2018-07-29T22.58-0c254e3/job.log
 (1/1) bonding.py:Bonding.test_cleanup;run-bdf8: PASS (12.69 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 12.79 s
JOB HTML   : /root/Upstream/Everglade/results/job-2018-07-29T22.58-0c254e3/results.html

# ip addr show bondtest
Device "bondtest" does not exist.

# 
```

**Comment by PraveenPenguin** [comment]
@manvanthar can you please review recent commit as well

File: io/net/bonding.py [hunk 2]
```diff
@@ -311,67 +301,68 @@ def bond_setup(self, arg1, arg2):
...
         self.ping_check()
+        self.error_check()
+
+    def test_run(self):
         self.bond_fail(self.mode)
         self.log.info("Mode %s OK", self.mode)
+        self.error_check()
 
-    def tearDown(self):
+    def test_cleanup(self):
```
Comment (PraveenPenguin) [inline io/net/bonding.py]: why this i think `tearDown` was good choice

Comment (narasimhan-v) [reply io/net/bonding.py]: @PraveenPenguin Agreed, but wanted bond remove as a seperate test.
Suggestions on handling are welcome..
Or give me a day to get a better flow :)

Comment (PraveenPenguin) [reply io/net/bonding.py]: sure as offline discussion we take this as enhancement , merging for now

Comment (narasimhan-v) [reply io/net/bonding.py]: @Sure, thanks.

**Pull request merged**